### PR TITLE
PMC Tools: Don't terminate on logged errors

### PR DESCRIPTION
**Pull request by bricelam**
Part of #8651

### CODE REVIEW

File: src/EFCore.Tools/tools/EntityFrameworkCore.psm1 [hunk 1]
```diff
@@ -679,6 +679,26 @@ function ShowConsole
...
+        # Call the internal API NuGet uses to display errors
+        $componentModel = Get-VSComponentModel
+        $powerConsoleWindow = $componentModel.GetService([NuGetConsole.IPowerConsoleWindow])
+        $bindingFlags = [Reflection.BindingFlags]::Instance -bor [Reflection.BindingFlags]::NonPublic
+        $activeHostInfo = $powerConsoleWindow.GetType().GetProperty('ActiveHostInfo', $bindingFlags).GetValue($powerConsoleWindow)
+        $internalHost = $activeHostInfo.WpfConsole.Host
+        $reportErrorMethod = $internalHost.GetType().GetMethod('ReportError', $bindingFlags, $null, [Exception], $null)
+        $exception = New-Object Exception $message
+        $reportErrorMethod.Invoke($internalHost, $exception)
+    }
```
Comment (smitpatel) [inline src/EFCore.Tools/tools/EntityFrameworkCore.psm1]: Magic :trollface:

File: src/EFCore.Tools/tools/EntityFrameworkCore.psm1 [hunk 2]
```diff
@@ -679,6 +679,26 @@ function ShowConsole
     $powerConsoleWindow.Show()
 }
 
+function WriteErrorLine($message)
+{
+    try
+    {
+        # Call the internal API NuGet uses to display errors
```
Comment (bricelam) [inline src/EFCore.Tools/tools/EntityFrameworkCore.psm1]: Older versions use `Red`; newer ones use `DarkRed`...

File: src/EFCore.Tools/tools/EntityFrameworkCore.psm1 (hunk 1 above)
Comment (bricelam) [inline src/EFCore.Tools/tools/EntityFrameworkCore.psm1]: lol, don't tell the NuGet team...

Comment (bricelam) [inline src/EFCore.Tools/tools/EntityFrameworkCore.psm1]: @ajcvickers Are you ok with this? If it ever breaks, we'll just fallback to `DarkRed`.

Comment (ajcvickers) [inline src/EFCore.Tools/tools/EntityFrameworkCore.psm1]: :shipit:

**Pull request merged**